Automattic/wp-calypso
# Monorepo: Move `bin/build-package.js` script to `calypso-build` package

#### Changes proposed in this Pull Request

* Use tabs (instead of spaces) to indent our `package.json` files (you might wanna look at this PR with [whitespace changes filtered](https://github.com/Automattic/wp-calypso/pull/31542/files?w=1)).
* Move `bin/build-package.js` script to `calypso-build` package
* Expose as that package's binary
* Invoke that binary from other packages' `package.json` files

I think it makes sense to co-locate this build script with the config files it is (or will be) based on. Arguably, it's more similar in spirit to those generic configs (which we are going to use to build packages) than to the Calypso-specific stuff at the root level. (It is also kind of a hard requirement for package isolation.)

Factored out of #31464 to make that PR smaller.

#### Testing instructions

```
npm run distclean
npm ci
npm start
```

and verify in your browser that Calypso actually works.

Then, build Jetpack blocks

```
npx lerna bootstrap --scope='@automattic/jetpack-blocks'
npx lerna run prepublishOnly --stream --scope='@automattic/jetpack-blocks'
```

and copy them to your local Jetpack Docker instance to test them there.

Finally, test package:

```
npm run test-packages
```
